crates/sui-framework/packages/sui-framework/sources/collectible/kiosk.move

```diff
@@ -181,21 +201,25 @@ module sui::kiosk {
 
     /// Place any object into a Kiosk.
     /// Performs an authorization check to make sure only owner can do that.
+    /// Makes sure a `TransferPolicy` exists for `T`, otherwise assets can be
+    /// locked in the `Kiosk` forever.
     public fun place<T: key + store>(
-        self: &mut Kiosk, cap: &KioskOwnerCap, item: T
-    ) {
+        self: &mut Kiosk, cap: &KioskOwnerCap, _policy: &TransferPolicy<T>, item: T
+    ): PlacedWitness<T> {
         assert!(object::id(self) == cap.for, ENotOwner);
         self.item_count = self.item_count + 1;
-        dof::add(&mut self.id, Item { id: object::id(&item) }, item)
+        dof::add(&mut self.id, Item { id: object::id(&item) }, item);
+        PlacedWitness {}
```

Comment:
Asking because it popped into my head: Is it worth including the placed item's `id` in the witness?

Reply:
Agreed with Ashok--unless the type in question is a singleton, I think we will always need to remember the ID/check it. Otherwise, it seems hard to avoid issues where someone "reuses" a witness for a different instance of `T` (+ hard to reason about what the consequences might be).

Reply:
ID in the witness won't be much help - we don't have the ID of the item in the TransferRequest so there's no way to compare it. TransferPolicy relies solely on type, and creating IDs everywhere can lead to "abuse of power" on the creator end.